include/cantera/thermo/RedlichKwongMFTP.h

```diff
@@ -1,380 +1,337 @@
-     * branch solution, it will return a -1 or a -2 instead of 1 or 2.  If it
-     * returns 0, then there is an error.
-     */
-    int NicholsSolve(double TKelvin, double pres, doublereal a, doublereal b,
-                     doublereal Vroot[3]) const;
-
-protected:
-    //! Form of the temperature parameterization
-    /*!
-     *  - 0 = There is no temperature parameterization of a or b
-     *  - 1 = The a_ij parameter is a linear function of the temperature
-     */
-    int m_formTempParam;
-
-    //! Value of b in the equation of state
-    /*!
-     *  m_b is a function of the temperature and the mole fraction.
-     */
-    doublereal m_b_current;
-
-    //! Value of a in the equation of state
-    /*!
-     *  a_b is a function of the temperature and the mole fraction.
-     */
-    doublereal m_a_current;
-
-    vector_fp a_vec_Curr_;
-    vector_fp b_vec_Curr_;
-
-    Array2D a_coeff_vec;
-
-    int NSolns_;
-
-    doublereal Vroot_[3];
-
-    //! Temporary storage - length = m_kk.
-    mutable vector_fp m_pp;
-
-    //! Temporary storage - length = m_kk.
-    mutable vector_fp m_tmpV;
-
-    // Partial molar volumes of the species
-    mutable vector_fp m_partialMolarVolumes;
-
-    //! The derivative of the pressure wrt the volume
-    /*!
-     * Calculated at the current conditions. temperature and mole number kept
-     * constant
-     */
-    mutable doublereal dpdV_;
-
-    //! The derivative of the pressure wrt the temperature
-    /*!
-     *  Calculated at the current conditions. Total volume and mole number kept
-     *  constant
-     */
-    mutable doublereal dpdT_;
-
-    //! Vector of derivatives of pressure wrt mole number
-    /*!
-     *  Calculated at the current conditions. Total volume, temperature and
-     *  other mole number kept constant
-     */
-    mutable vector_fp dpdni_;
-
-public:
-    //! Omega constant for a -> value of a in terms of critical properties
-    /*!
-     *  this was calculated from a small nonlinear solve
-     */
-    static const doublereal omega_a;
-
-    //! Omega constant for b
-    static const doublereal omega_b;
-
-    //! Omega constant for the critical molar volume
-    static const doublereal omega_vc;
-};
-}
-
-#endif
+//! @file RedlichKwongMFTP.h
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#ifndef CT_REDLICHKWONGMFTP_H
+#define CT_REDLICHKWONGMFTP_H
+
+#include "MixtureFugacityTP.h"
+#include "cantera/base/Array.h"
+
+namespace Cantera
+{
+/**
+ * Implementation of a multi-species Redlich-Kwong equation of state
+ *
+ * @ingroup thermoprops
+ */
+class RedlichKwongMFTP : public MixtureFugacityTP
+{
+public:
+    //! @name Constructors and Duplicators
+    //! @{
+
+    //! Base constructor.
+    RedlichKwongMFTP();
+
+    //! Construct and initialize a RedlichKwongMFTP object directly from an
+    //! ASCII input file

```

Comment:
Well, I guess this is probably copied from somewhere else. Anyways, can we fix it up? Thanks :smiley:

Reply:
Corrected.